tests/test_callbacks.py

```diff
@@ -146,3 +147,33 @@ def test_async_async_callbacks():
     t = Thread(target=test_async_callbacks)
     t.start()
     t.join()
+
+
+def test_callback_num_times(capsys):
```

Comment:
FWIW `pytest` always captures the output, and only reports on error, right?
Hence the recommendation for stuff like `--capture=no` here: https://github.com/pybind/pybind11/blob/f676782becc5bcd30c56b4da1ff1026db61bcce3/.github/CONTRIBUTING.md#configuration-options

So a defect in some contexts, but prolly not here? (i.e. it's probably good to let it run since it's short?)

Reply:
I removed `capsys` entirely. Simply running with `-k test_callback_num_times -s` is easy and best.

Reply:
Okay. I think we should probably add a mark for this eventually, but not for just one item yet. @EricCousineau-TRI I believe @rwgk disabled capturing explicitly so that this would print out before, but he's removed that now.

Reply:
Ah, derp :facepalm: Thanks for explaining!